### PR TITLE
Move updateCodeLenses logic to AgentFixupControls

### PR DESCRIPTION
**Pull request by mkondratek**
Fixes https://github.com/sourcegraph/jetbrains/issues/2197.

## Test plan
Try to reproduce https://github.com/sourcegraph/jetbrains/issues/2197.

## Changelog
Move updateCodeLenses from agent.ts to AgentFixupControls.ts

### CODE REVIEW

File: vscode/src/non-stop/codelenses/provider.ts [hunk 1]
```diff
@@ -157,8 +157,9 @@ export class FixupCodeLenses implements vscode.CodeLensProvider, FixupControlApp
             this.removeLensesFor(task)
             return
         }
-        this.taskLenses.set(task, getLensesForTask(task))
-        this.notifyCodeLensesChanged()
+        const value = getLensesForTask(task)
```
Comment (pkukielka) [inline vscode/src/non-stop/codelenses/provider.ts:160]: Please name it bettter

File: agent/src/AgentFixupControls.ts [hunk 2]
```diff
@@ -60,6 +61,59 @@ export class AgentFixupControls extends FixupCodeLenses {
...
 
+    override notifyCodeLensesChanged(uri: vscode.Uri, codeLenses: vscode.CodeLens[]) {
+        super.notifyCodeLensesChanged(uri, codeLenses)
+        void this.updateCodeLenses(uri, codeLenses)
+    }
+
+    private async updateCodeLenses(uri: vscode.Uri, codeLenses: vscode.CodeLens[]): Promise<void> {
+        // VS Code supports icons in code lenses, but we cannot render these through agent.
+        // We need to strip any icons from the title and provide those seperately, so the client can decide how to render them.
+        const agentLenses = codeLenses.map(lens => {
```
Comment (pkukielka) [inline agent/src/AgentFixupControls.ts:72]: You could make this more concise:

```
const agentLenses = lenses.map(lens => ({
    ...lens,
    command: lens.command
        ? {
              ...lens.command,
              title: this.splitIconsFromTitle(lens.command.title),
          }
        : undefined,
}))
```

File: agent/src/agent.ts [hunk 3]
```diff
@@ -1529,78 +1524,6 @@ export class Agent extends MessageHandler implements ExtensionClient {
...
-    private async updateCodeLenses(): Promise<void> {
-        const uri = this.workspace.activeDocumentFilePath
-        if (!uri) {
-            return
-        }
-        const document = this.workspace.getDocument(uri)
-        if (!document) {
-            return
-        }
-        this.codeLensToken.cancel()
```
Comment (pkukielka) [inline agent/src/agent.ts:1564]: Great, is is much simpler now :)

File: agent/src/AgentFixupControls.ts (hunk 2 above)
Comment (mkondratek) [reply agent/src/AgentFixupControls.ts]: done

File: agent/src/edit.test.ts [hunk 4]
```diff
@@ -35,7 +35,8 @@ describe('Edit', () => {
         })
         await client.taskHasReachedAppliedPhase(task)
         const lenses = client.codeLenses.get(uri.toString()) ?? []
-        expect(lenses).toHaveLength(2)
```
Comment (mkondratek) [inline agent/src/edit.test.ts:38]: It looks like this was a bug in the tests. I am surprised we lived with it till today 😅

Comment (pkukielka) [reply agent/src/edit.test.ts:38]: How is that possible? Tests were not executing?

**Review by pkukielka — APPROVED** [review]
LGTM

**Pull request merged**